crates/bevy_math/src/curves/fixed.rs

```diff
@@ -0,0 +1,121 @@
+use crate::{
+    curves::{Curve, CurveCursor},
+    interpolation::Lerp,
+};
+
+// TODO: impl Serialize, Deserialize
+/// Curve with evenly spaced keyframes, in another words a curve with a fixed frame rate.
+///
+/// This curve maintains the faster sampling rate over a wide range of frame rates, because
+/// it doesn't rely on keyframe cursor. As a downside, it will have a bigger memory foot print.
+#[derive(Default, Debug)]
+pub struct CurveFixed<T> {
+    frame_rate: f32,
+    /// Negative number of frames before the curve starts
+    negative_offset: f32,
+    pub keyframes: Vec<T>,
+}
+
+impl<T: Clone> Clone for CurveFixed<T> {
+    fn clone(&self) -> Self {
+        Self {
+            frame_rate: self.frame_rate,
+            negative_offset: self.negative_offset,
+            keyframes: self.keyframes.clone(),
+        }
+    }
+}
+
+impl<T> CurveFixed<T> {
+    pub fn from_keyframes(frame_rate: f32, offset: isize, keyframes: Vec<T>) -> Self {
```

Comment:
I would like some guidance on choosing the value of `offset` here. I have no idea where to begin and this is part of the public API.

Reply:
Probably most fitting in `set_offset`?

Reply:
If I name it `frame_offset`, does make it more clearer?

Reply:
@NathanSWard here

Reply:
`frame_offset` is much clearer to me :)